server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1011,6 +1016,11 @@ static List<SearchShardIterator> getRemoteShardsIteratorFromPointInTime(
                 remoteShardIterators.add(shardIterator);
             }
         }
+
+        assert searchContextIdForNodeMap.values()
+            .stream()
+            .filter(searchContextIdForNode -> searchContextIdForNode.getClusterAlias() != null)
```

Comment:
I guess we could just do a `noneMatch` here ? :)

Reply:
yep